packages/steps/src/BuildStepContext.ts

```diff
@@ -2,30 +2,53 @@ import os from 'os';
 import path from 'path';
 
 import { bunyan } from '@expo/logger';
+import { v4 as uuidv4 } from 'uuid';
 
 import { BuildStep } from './BuildStep.js';
 import { parseOutputPath } from './utils/template.js';
 import { BuildStepRuntimeError } from './errors.js';
 import { BuildRuntimePlatform } from './BuildRuntimePlatform.js';
+import { BuildStepEnv } from './BuildStepEnv.js';
 
-export class BuildStepContext {
+export interface BuildContextProvider {
```

Comment:
Not really sure about the name here too. Context provider suggests that it will provide a context, but it's in fact providing values that context needs internally.

I could drop a provider here, and name it sth like e.g. ExternalBuildContext or maybe BuildContextImpl or sth like that, but this would mean we 3 separate context objects here, which is a bit messy.